File: Backend/C#/SuperHero.Service/BaseHeroService.cs
using SuperHero.Domain.Behavior.Service;
using SuperHero.Repository.Behavior;

namespace SuperHero.Service
{
    public class BaseHeroService : IBaseHeroService
    {
        private readonly IBaseHeroPersister _baseHeroPersister;
        public BaseHeroService(IBaseHeroPersister baseHeroPersister)
        {
            _baseHeroPersister = baseHeroPersister;
        }

        public async Task CreateBaseHero(IEnumerable<Domain.Model.BaseHero> baseHeroCollection)
        {
            _ = await _baseHeroPersister.CreateBaseHero(baseHeroCollection);
        }
    }
}

File: Backend/Typescript/src/services/BaseHeroService.ts
import { inject, injectable } from "tsyringe";
import type { IBaseHeroService } from "../domain/behavior/service/base-hero-service.interface.js";
import type { BaseHero } from "../domain/entities/base/base-hero.js";
import type { IBaseHeroPersister } from "../domain/behavior/repository/baase-hero-persister.interface.js";

@injectable()
export class BaseHeroService implements IBaseHeroService {
    constructor(@inject("IBaseHeroPersister") private baseHeroPersister: IBaseHeroPersister) { }

    async createBaseHero(baseHeroCollection: BaseHero[]): Promise<void> {
        try {
            await this.baseHeroPersister.createBaseHero(baseHeroCollection);
        } catch (error) {
            console.error('Error creating base heroes:', error);
            throw error;
        }
    }
}